task/sast-unicode-check/0.1/sast-unicode-check.yaml

```diff
@@ -0,0 +1,281 @@
+                echo "Message: Succeed to run find-unicode-control command" >&2
+            else
+                # Failed to run the command
+                echo "Error: failed to run find unicode control command" >&2
+                return 1
+            fi
+
+            # Translate the output format
+            TRANSLATE_EXIT_CODE=0
+            sed -i raw_sast_unicode_check_out.txt -E -e 's|(.*:[0-9]+)(.*)|\1: warning:\2|' -e 's|^|Error: UNICONTROL_WARNING:\n|' || TRANSLATE_EXIT_CODE=$?
+            if [[ "${TRANSLATE_EXIT_CODE}" -eq 0 ]]; then
+                echo "Message: Succeed to translate the unicontrol output format" >&2
+            else
+                echo "Error: failed to translate the unicontrol output format" >&2
+                return 1
+            fi
+        }
+
+        # Process all results as configured with CSGERP_OPTS
+        process_all_results()
+        {
+            CSGERP_OPTS=(
+                --mode=json
+                --remove-duplicates
+                --embed-context=3
+                --set-scan-prop="${SCAN_PROP}"
+                --strip-path-prefix="${SOURCE_CODE_DIR}"/source/
+            )
+            # In order to generate csdiff/v1, we need to add the whole path of the source code as sast-unicode-check only provides an URI to embed the context
+            csgrep "${CSGERP_OPTS[@]}" raw_sast_unicode_check_out.txt > processed_sast_unicode_check_out.json
+
+            csgrep --mode=evtstat processed_sast_unicode_check_out.json
+        }
+
+        # Filter known false positive
+        csfilter_kfp()
+        {
+            echo "Message: Filtering false positives in results files using ${KFP_GIT_URL}..." >&2
+
+            # Build initial csfilter-kfp command
+            csfilter_kfp_cmd="csfilter-kfp --verbose --kfp-git-url=${KFP_GIT_URL}"
+
+            # Append --project-nvr option if PROJECT_NVR is set
+            if [[ -n "${PROJECT_NVR}" ]]; then
+                csfilter_kfp_cmd+=" --project-nvr="${PROJECT_NVR}""
+            fi
+
+            # Append --record-excluded option if RECORD_EXCLUDED is true
+            if [[ "${RECORD_EXCLUDED}" == "true" ]]; then
+                csfilter_kfp_cmd+=" --record-excluded=raw-excluded-findings.json"
+            fi
+
+            KFP_EXIT_CODE=0
+            ${csfilter_kfp_cmd[@]} processed_sast_unicode_check_out.json > sast_unicode_check_out.json || KFP_EXIT_CODE=$?
+            if [ "${KFP_EXIT_CODE}" -eq 0 ]; then
+                echo "Message: Succeed to filter known false positives" >&2
+            else
+                echo "Error: failed to filter known false positives" >&2
+                return 1
+            fi
+
+            if [ -f "raw-excluded-findings.json" ]; then
+                csgrep --mode=json --strip-path-prefix="${SOURCE_CODE_DIR}"/source/ raw-excluded-findings.json > excluded-findings.json
+            fi
+        }
+
+        generate_sarif_report()
```

Comment:
The generation of the SARIF file is just one line. Do we want to call this function generate_sraif_report? The rest are parsing for the task output

Reply:
To be honest, I am not sure if the use of these shell functions really improves anything.  They are all executed one by one in the same order as they are defined.  They do not use any local variables and the few occurrences of `return 1` can be replaced by `exit 1` without any observable change in behavior.  If we inline these functions, the whole script will be shorter with fewer nesting levels, and we can skip the discussion about suitable function names.

Reply:
I have removed these functions, only using command lines